src/veh_interact.h

```diff
@@ -115,6 +116,14 @@ class veh_interact
         void display_details( const vpart_info *part );
         size_t display_esc( WINDOW *w );
 
+        /**
+          * Display overview of parts
+          * @param enable used to determine if a part can be selected
+          * @param action callback run when a part is selected
+          */
+        void overview( std::function<bool( const vehicle_part &pt )> enable = {},
```

Comment:
`veh_interact.h` should explicitly include `<functional>` if you plan to use `std::function`. However, I don't particularly like the idea of passing functions here. You use them in one place only to customize reactors and fuel tanks, and nowhere else. This seems to be an overkill, and the whole thing can be simplified.